test/performance/suites/external_segment.rb

```diff
@@ -39,6 +39,17 @@ def test_external_request
     stop_server io_server
   end
 
+  def test_external_request_in_thread
```

Comment:
How do these performance tests look with the latest updates?

Reply:
I haven't run them yet! Thanks for the reminder, I definitely want to do that.

Reply:
Completed the perf suite tests for this PR